src/storage/src/controller/hosts.rs

```diff
@@ -125,20 +123,17 @@ where
     }
 
     /// Provisions a storage host for the storage object with the specified ID.
-    ///
-    /// If `remote_addr` is `Some`, then the specified storage host is used.
-    /// Otherwise, a storage host is assigned automatically.
+    /// If the storage host is managed, this will ensure that the backing orchestrator
+    /// allocates resources; if not, it will ensure that the orchestrator will
+    /// release any resources. (For 'remote' storaged instances, the user is required
```

Comment:
In what cases we release resources?

Reply:
I think the only case is when we're changing a source from managed to remote: in that case we need to decommission the k8s service or we'll leak it.

Reply:
Made the comment more explicit on this point!